src/Mono.Android/Android.Runtime/JNIEnv.cs

```diff
@@ -162,33 +164,51 @@ static unsafe void RegisterJniNatives (IntPtr typeName_ptr, int typeName_len, In
 			if (methods.Length == 0)
 				return;
 
-			JniNativeMethod[] natives = new JniNativeMethod [methods.Length-1];
-			for (int i = 0; i < methods.Length; ++i) {
-				string method = methods [i];
-				if (string.IsNullOrEmpty (method))
-					continue;
-				string[] toks = methods [i].Split (new[]{':'}, 4);
-				Delegate callback;
-				if (toks [2] == "__export__") {
-					var mname = toks [0].Substring (2);
-					var minfo = type.GetMethods (BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance | BindingFlags.Static).Where (m => m.Name == mname && JavaNativeTypeManager.GetJniSignature (m) == toks [1]).FirstOrDefault ();
-					if (minfo == null)
-						throw new InvalidOperationException (String.Format ("Specified managed method '{0}' was not found. Signature: {1}", mname, toks [1]));
-					callback = CreateDynamicCallback (minfo);
-				} else {
-					GetCallbackHandler connector = (GetCallbackHandler) Delegate.CreateDelegate (typeof (GetCallbackHandler), 
-						toks.Length == 4 ? Type.GetType (toks [3], true) : type, toks [2]);
-					callback = connector ();
+			bool registered = false;
+#if JAVA_INTEROP
```

Comment:
We don't need to add more uses of `#if JAVA_INTEROP`; it's not going away now. :-)

Reply:
That's good to know :-)